### ISSUE TITLE
First Conversion Very Slow

### ISSUE THREAD

**Issue by sshaw** [Shopify/measured#103]
This seems to be the case with `Weight` and other conversions too but fl oz to ml is the slowest so far:
```
irb [2.4.0] (tmp)$ v = Measured::Volume.parse("123 fl_oz")
=> #<Measured::Volume: 123 #<Measured::Unit: oz (fl_oz, imp_fl_oz, imperial_fluid_ounce, imperial_fluid_ounces) 1/160 gal>>
irb [2.4.0] (tmp)$ Benchmark.measure { v.convert_to("ml") }.real
=> 2.460711195017211
```

**Comment by kmcphillips** [comment]
```ruby
require "ruby-prof"
require "measured"

volume = Measured::Volume.parse("123 fl_oz")

RubyProf.start
volume.convert_to("ml")
result = RubyProf.stop

printer = RubyProf::FlatPrinter.new(result)
printer.print(STDOUT)
```
Produces
```
 %self      total      self      wait     child     calls  name
  4.75      5.254     0.262     0.000     4.992   398924   Measured::ConversionTable#find_direct_conversion
```
Yeah we clearly had a serious performance regression here.

**Comment by thegedge** [comment]
Ah yes, looks like we lazily build the conversion table. Definitely shouldn't do that!

**Comment by sshaw** [comment]
@kmcphillips, I'm curious, given #102, how does your profiling code run?

**Comment by kmcphillips** [comment]
@sshaw I basically use `pry` everywhere. I'm sure I had it required into my testing environment there. We have it required in tests and it's in all my base gemsets.

#104 Fixes it by explicitly requiring it.

@javierhonduco did some amazing work speeding it up by cutting some unnecessary operations in #105, but I'm still convinced that that tree traversal is broken if it's doing hundreds of thousands of iterations. I'm going to keep digging.

**Comment by kmcphillips** [comment]
Though already it's gone from `398924` to `132070` calls of `find_direct_conversion` and is almost four times faster on my machine. But still can be improved.

**Comment by javierhonduco** [comment]
@sshaw We just released [`measured` v2.4.0](https://github.com/Shopify/measured/releases/tag/v2.4.0) which should make conversions way faster, although there's still room for improvement! 😄
